caphandler/addresspools.go

```diff
@@ -37,19 +36,17 @@ func GetAddressPoolCollection(ctx iris.Context) {
 	fabricID := ctx.Params().Get("id")
 	// get all switches which are store under that fabric
 
-	if _, err := capmodel.GetFabric(fabricID); err != nil {
+	var members = []*model.Link{}
+	addressPools, err := capmodel.GetAllAddressPools(fabricID)
+	if err != nil {
 		errMsg := fmt.Sprintf("failed to fetch AddressPool data for uri %s: %s", uri, err.Error())
 		createDbErrResp(ctx, err, errMsg, []interface{}{"AddressPool", uri})
 		return
 	}
-
-	var members = []*model.Link{}
-	for addressPoolOID, addressPooldData := range capdata.AddressPoolDataStore {
-		if addressPooldData.FabricID == fabricID {
-			members = append(members, &model.Link{
-				Oid: addressPoolOID,
-			})
-		}
+	for addressPoolOID := range addressPools {
+		members = append(members, &model.Link{
```

Comment:
why did we remove the check for fabricID?

Reply:
Key set name contains the fabric ID using which we fetch the address pool data, hence additional DB call to check the fabric ID was removed.

Reply:
will that handle multiple fabric IDs ?

Reply:
yes, multiple unique fabric IDs is supported